catalog/ftp/src/main/java/ddf/catalog/ftp/FtpServerManager.java

```diff
@@ -252,7 +252,8 @@ private void waitForConnections() {
           break;
         }
       } catch (InterruptedException e) {
-        Thread.interrupted();
+        Thread.interrupted(); // <<<<<<<< Do we still need this line?
+        Thread.currentThread().interrupt();
```

Comment:
❗️ See my previous comment about resetting the interrupted flag within a loop. It needs to be done outside of the loop.